### PR TITLE
Fixes #25502 - Better error handling for repeated uploads

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: lib/foreman_openscap/helper.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module ForemanOpenscap::Helper
   def self.get_asset(cname, policy_id)
-    asset = find_host_by_name_or_uuid(cname).get_asset
+    asset = find_host_by_name_or_uuid(cname)&.get_asset
```
[the remark this answers is not in the document]
Comment (xprazak2) [reply lib/foreman_openscap/helper.rb]: It is safe navigation operator.

**Comment by lzap** [comment]
@ares @xprazak2 seems like an important bug for 6.4 I offer help with review and testing. Will need assistance setting up dev/test environment tho.

File: app/models/foreman_openscap/arf_report.rb [hunk 2]
```diff
@@ -105,16 +105,17 @@ def rules_count
...
-                                       :reported_at => Time.at(params[:date].to_i),
-                                       :status => params[:metrics],
-                                       :metrics => params[:metrics],
-                                       :openscap_proxy => proxy)
+        arf_report = ArfReport.create(:host => asset.host,
+                                      :reported_at => Time.at(params[:date].to_i),
+                                      :status => params[:metrics],
+                                      :metrics => params[:metrics],
+                                      :openscap_proxy => proxy)
+        return arf_report unless arf_report.persisted?
```
Comment (lzap) [inline app/models/foreman_openscap/arf_report.rb]: Can you explain this? Now it always return the newly created report so the rest of the method is completely ignored, is this intentional?

File: test/functional/api/v2/compliance/arf_reports_controller_test.rb [hunk 3]
```diff
@@ -93,6 +93,40 @@ class Api::V2::Compliance::ArfReportsControllerTest < ActionController::TestCase
...
+                                     :date => date.to_i,
+                                     :openscap_proxy_name => @proxy.name),
+         :session => set_session_user
+    assert_response :unprocessable_entity
+    res = JSON.parse(@response.body)
+    msg = "Could not find host identified by: #{cname}"
+    assert_equal msg, res["errors"]
+  end
+
+  test "should not create report when host is missing" do
```
Comment (lzap) [inline test/functional/api/v2/compliance/arf_reports_controller_test.rb]: Nitpick: very same test name, it is probably a different one.

**Comment by lzap** [comment]
Also relevant test failures.

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 4]
```diff
@@ -75,6 +75,16 @@ def download_html
 
         private
 
+        def respond_for_report(arf_report)
+          if arf_report.nil?
+            upload_fail (_("Policy with id %s not found.") % params[:policy_id])
```
[the remark this answers is not in the document]
Comment (ares) [reply app/controllers/api/v2/compliance/arf_reports_controller.rb]: I suppose this does not like the space between upload_fail and parenthesis

File: lib/foreman_openscap/helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ares) [reply lib/foreman_openscap/helper.rb]: if rucobop is wrong, we need to disable the cop, otherwise tests will start failing on master after this is merged

**Comment by xprazak2** [comment]
I bumped the target ruby version for rubocop + test fixes.

**Comment by lzap** [comment]
Rubocop is still failing, but I am going to start testing as tests are passing.

**Comment by ares** [comment]
@xprazak2 mind to rebase after rubocop config was updated?

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 5]
```diff
@@ -75,6 +75,16 @@ def download_html
 
         private
 
+        def respond_for_report(arf_report)
+          if arf_report.nil?
+            upload_fail(_("Policy with id %s not found.") % params[:policy_id])
+          elsif arf_report.new_record?
+            upload_fail arf_report.errors.full_messages.to_sentence
+          else
+            render :json => { :result => :ok, :id => arf_report.id.to_s }
```
[the remark this answers is not in the document]
Comment (xprazak2) [reply app/controllers/api/v2/compliance/arf_reports_controller.rb]: Yes, to keep everything small-caps.

**Comment by lzap** [comment]
There is one test failure however, not sure:

```
--- expected
+++ actual
@@ -1 +1 @@
-"http://localhost:8443"
+"http://proxy.example.com:8443"
 (Minitest::Assertion)
/var/lib/workspace/workspace/foreman_openscap-pull-request/database/sqlite3/label/fast/ruby/2.5/plugin/test/unit/scap_content_test.rb:32   
```

**Comment by xprazak2** [comment]
[test foreman_openscap]

**Comment by ares** [comment]
Thanks @xprazak2 and @lzap, merging!

**Pull request merged**